### PR TITLE
Add Portieris container resource requests, remove readiness probe

### PR DESCRIPTION
**Pull request by lucaspalm**

### CODE REVIEW

**Review by rtheis — CHANGES_REQUESTED** [review]
Please support customizing resources and probes like is done for KMS.

File: assets/kube-apiserver/kube-apiserver-deployment.yaml [hunk 1]
```diff
@@ -254,23 +254,24 @@ spec:
...
-          initialDelaySeconds: 10
-          timeoutSeconds: 10
-        readinessProbe:
-          httpGet:
-            port: 8000
-            path: "/health/readiness"
-            scheme: HTTPS
-          initialDelaySeconds: 10
-          timeoutSeconds: 10
+          initialDelaySeconds: 120
```
Comment (rtheis) [inline assets/kube-apiserver/kube-apiserver-deployment.yaml:273]: Please add the probe configuration support as well.

**Review by rtheis — CHANGES_REQUESTED** [review]

File: assets/kube-apiserver/kube-apiserver-deployment.yaml [hunk 2]
```diff
@@ -254,23 +254,28 @@ spec:
...
+{{ if .PortierisContainerResources }}
+        resources:{{ range .PortierisContainerResources }}{{ range .ResourceRequest }}
+          requests: {{ if .CPU }}
+            cpu: {{ .CPU }}{{ end }}{{ if .Memory }}
+            memory: {{ .Memory }}{{ end }}{{ end }}{{ range .ResourceLimit }}
+          limits: {{ if .CPU }}
+            cpu: {{ .CPU }}{{ end }}{{ if .Memory }}
+            memory: {{ .Memory }}{{ end }}{{ end }}{{ end }}
+{{ end }}
+{{ if .KubeAPIServerSecurity }}
```
Comment (rtheis) [inline assets/kube-apiserver/kube-apiserver-deployment.yaml:266]: Please leave the security context work to https://github.com/openshift/ibm-roks-toolkit/pull/114.

**Review by rtheis — CHANGES_REQUESTED** [review]

File: assets/kube-apiserver/kube-apiserver-deployment.yaml [hunk 3]
```diff
@@ -250,35 +250,59 @@ spec:
       - name: portieris
         image: {{ .PortierisImage }}
         imagePullPolicy: IfNotPresent
+        command:
+          - "portieris"
+        args:
+          - "--kubeconfig=/etc/openshift/kubeconfig/kubeconfig"
```
Comment (lucaspalm) [inline assets/kube-apiserver/kube-apiserver-deployment.yaml:256]: Support for this flag is currently under review in https://github.com/IBM/portieris/pull/218

File: assets/kube-apiserver/kube-apiserver-deployment.yaml [hunk 4]
```diff
@@ -250,35 +250,59 @@ spec:
...
+          - "--alsologtostderr"
+          - "-v=4"
+          - "2>&1"
```
Comment (rtheis) [inline assets/kube-apiserver/kube-apiserver-deployment.yaml:259]: Do you really need to redirect output to capture it in the container logs?

File: assets/kube-apiserver/kube-apiserver-deployment.yaml [hunk 5]
```diff
@@ -250,35 +250,59 @@ spec:
       - name: portieris
         image: {{ .PortierisImage }}
         imagePullPolicy: IfNotPresent
+        command:
+          - "portieris"
+        args:
+          - "--kubeconfig=/etc/openshift/kubeconfig/kubeconfig"
+          - "--alsologtostderr"
+          - "-v=4"
```
Comment (rtheis) [inline assets/kube-apiserver/kube-apiserver-deployment.yaml:258]: This level of verbosity seems a bit too high for the default value.

File: assets/kube-apiserver/kube-apiserver-deployment.yaml [hunk 6]
```diff
@@ -250,35 +250,59 @@ spec:
...
+{{- $probe := .PortierisLivenessProbe }}
         livenessProbe:
           httpGet:
-            port: 8000
-            path: "/health/liveness"
-            scheme: HTTPS
-          initialDelaySeconds: 10
-          timeoutSeconds: 10
-        readinessProbe:
+            scheme: {{ or $probe.HttpGet.Scheme "HTTP" }}
```
Comment (rtheis) [inline assets/kube-apiserver/kube-apiserver-deployment.yaml:279]: Shouldn't the default be `HTTPS`?

**Review by rtheis — CHANGES_REQUESTED** [review]

File: assets/kube-apiserver/kube-apiserver-deployment.yaml [hunk 7]
```diff
@@ -250,35 +250,56 @@ spec:
...
-          timeoutSeconds: 10
-        env:
-        - name: KUBECONFIG
-          value: /etc/kubernetes/secret/kubeconfig
+          initialDelaySeconds: 120
+          periodSeconds: 300
+          successThreshold: 1
+          failureThreshold: 3
+          timeoutSeconds: 160
+{{ end }}
```
Comment (rtheis) [inline assets/kube-apiserver/kube-apiserver-deployment.yaml:295]: Does this need to be `{{- end }}`?

File: assets/kube-apiserver/kube-apiserver-deployment.yaml [hunk 8]
```diff
@@ -250,35 +250,56 @@ spec:
...
         volumeMounts:
-        - mountPath: /etc/kubernetes/secret/
-          name: secret
+        - mountPath: /etc/openshift/kubeconfig
+          name: kubeconfig
```
Comment (rtheis) [inline assets/kube-apiserver/kube-apiserver-deployment.yaml:298]: You need to update the volumes to ensure this mount is available when `PortierisEnabled`.  It is only available for `ROKSMetricsImage` at the moment.

**Review by rtheis — CHANGES_REQUESTED** [review]

**Comment by lucaspalm** [comment]
@rtheis All your comments/concerns have been addressed so far.

File: assets/kube-apiserver/kube-apiserver-deployment.yaml [hunk 9]
```diff
@@ -250,35 +250,56 @@ spec:
...
           httpGet:
-            port: 8000
-            path: "/health/liveness"
-            scheme: HTTPS
-          initialDelaySeconds: 10
-          timeoutSeconds: 10
-        readinessProbe:
+            scheme: {{ or $probe.HttpGet.Scheme "HTTPS" }}
+            port: {{ or $probe.HttpGet.Port 8000 }}
+            path: {{ or $probe.HttpGet.Path "health/liveness" }}
```
Comment (rtheis) [inline assets/kube-apiserver/kube-apiserver-deployment.yaml:278]: Sorry I missed this on my previous reviews, but the default should be `/health/liveness`.  The leading slash is missing.

**Review by rtheis — CHANGES_REQUESTED** [review]

**Review by rtheis — APPROVED** [review]
/lgtm

**Comment by rtheis** [comment]
@csrwng ptal

**Comment by csrwng** [comment]
I'm seeing:
```
t % ./bin/ibm-roks render                                                                       
panic: template: template:319:7: executing "template" at <.ROKSMetricsImage>: ROKSMetricsImage has arguments but cannot be invoked as function

goroutine 1 [running]:
github.com/openshift/ibm-roks-toolkit/pkg/render.(*renderContext).substituteParams(0xc0010be000, 0x1838e00, 0xc000146900, 0x1b7d39b, 0x2d, 0x45, 0x1bac44f, 0x49, 0x1bafe07)
        /home/cewong/code/ibm-roks-toolkit/pkg/render/rendercontext.go:68 +0x30b
github.com/openshift/ibm-roks-toolkit/pkg/render.(*renderContext).renderManifests(0xc0010be000, 0xc001010001, 0x1838e00)
        /home/cewong/code/ibm-roks-toolkit/pkg/render/rendercontext.go:39 +0x2eb
github.com/openshift/ibm-roks-toolkit/pkg/render.RenderClusterManifests(0xc000146900, 0xc000376640, 0x32, 0xc00011a2a0, 0x2c, 0xc000dc0001, 0xc000dcfd88, 0xc000dcfd88)
        /home/cewong/code/ibm-roks-toolkit/pkg/render/manifests.go:23 +0x139
github.com/openshift/ibm-roks-toolkit/pkg/cmd/render.(*RenderManifestsOptions).Run(0xc00032e400, 0x1b4089c, 0x4)
        /home/cewong/code/ibm-roks-toolkit/pkg/cmd/render/manifests.go:48 +0x14c
github.com/openshift/ibm-roks-toolkit/pkg/cmd/render.NewRenderManifestsCommand.func1(0xc000e88840, 0x29f7010, 0x0, 0x0)
        /home/cewong/code/ibm-roks-toolkit/pkg/cmd/render/manifests.go:26 +0x2e
github.com/spf13/cobra.(*Command).execute(0xc000e88840, 0x29f7010, 0x0, 0x0, 0xc000e88840, 0x29f7010)
        /home/cewong/code/ibm-roks-toolkit/vendor/github.com/spf13/cobra/command.go:846 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xc000e88580, 0xc000dcff70, 0x1, 0x1)
        /home/cewong/code/ibm-roks-toolkit/vendor/github.com/spf13/cobra/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /home/cewong/code/ibm-roks-toolkit/vendor/github.com/spf13/cobra/command.go:887
main.main()
        /home/cewong/code/ibm-roks-toolkit/cmd/ibm-roks/main.go:12 +0xb3
```
which I don't get running render with the master branch
I suspect it's being caused by:
https://github.com/openshift/ibm-roks-toolkit/pull/113/files#diff-8b7c5159d658cb2b12ea1b22b94ce9efc30358c9d47a47c6fde717ac9c3a7ca1R319

**Comment by lucaspalm** [comment]
The "or" operator must not be acceptable there?  Whats the proper way to do that?

**Comment by csrwng** [comment]
@lucaspalm I think the right syntax is `if or .ROKSMetricsImage .PortierisEnabled` (since `or` is considered a function with 2 params)

**Comment by lucaspalm** [comment]
@csrwng Yep that's what I changed it too.  See if that works now...

**Comment by csrwng** [comment]
@lucaspalm if you think it's ready, remove the wip tag

**Comment by lucaspalm** [comment]
@csrwng Let me finish up some testing with it first before I am happy removing the tag.

**Review by rtheis — APPROVED** [review]
/lgtm

**Comment by lucaspalm** [comment]
@rtheis FYI the latest push was to add back in the logging flags.  Without them, there are basically no helpful logs...

@csrwng My testing looks good.  This is ready when you and are.

**Review by rtheis — APPROVED** [review]
/lgtm

**Comment by lucaspalm** [comment]
I guess im not sure why the tests are now failing...
```
error: some steps failed:
  * could not run steps: step src failed: the build src failed after 14s with reason FetchImageContentFailed: Failed to extract image content. 
```
Compared to the previous passing build, all I did was rebase and add a couple starting command args.

Locally, `make verify` looks good.

**Pull request merged**